src/engraving/dom/edit.cpp

```diff
@@ -6427,17 +6437,7 @@ void Score::undoRemoveElement(EngravingItem* element, bool removeLinked)
             if (e->explicitParent() && e->explicitParent()->isSystem()) {
                 e->setParent(0);   // systems will be regenerated upon redo, so detach
             }
-        }
-    }
-
-    if (element->isMeasureRepeat()) {
```

Comment:
Why did you move that piece of code?

Reply:
I deleted the elements immediately in `doUndoRemoveElement` before I found `deleteLater`.  This is unnecessary now, I'll move it back.